packages/dds/tree/src/simple-tree/node-kinds/object/objectNode.ts

```diff
@@ -553,13 +578,16 @@ export function objectSchema<
 		public get [typeSchemaSymbol](): Output {
 			return CustomObjectNode.constructorCached?.constructor as unknown as Output;
 		}
-		public static readonly [privateDataSymbol] = createTreeNodeSchemaPrivateData(
-			this,
-			Array.from(
-				flexKeyMap.values(),
-				({ schema }) => normalizeFieldSchema(schema).annotatedAllowedTypes,
-			),
-		);
+
+		public static get [privateDataSymbol](): TreeNodeSchemaPrivateData {
+			return (privateData ??= createTreeNodeSchemaPrivateData(
```

Comment:
I had to convert these into getters instead of just normal properties since they now capture `this`, and we need them to capture the most derived `this` for subclassable schema.